### PR TITLE
Remove default parameters from cudf::detail::sort function declarations

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Removes the default parameter declarations from `cudf::detail::sort`, `cudf::detail::sorted_order`, `cudf::detail::sort_by_key`,  `cudf::detail::stable_sorted_order`, and `cudf::detail::stable_sort_by_key` as well as `cudf::detail::quantile`, `cudf::detail::quantiles` and `cudf::detail::percentile_approx`
Reference comment: https://github.com/rapidsai/cudf/pull/12237#issuecomment-1329538871
Also fixes incorrect stream parameter passed within a `quantiles.cu` detail function.
No function has been added, modified, or changed.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by harrism — APPROVED** [review]

File: cpp/src/reductions/reductions.cpp [hunk 1]
```diff
@@ -70,18 +70,22 @@ struct reduce_dispatch_functor {
...
         return get_element(*col_ptr, 0, stream, mr);
       }
       case aggregation::QUANTILE: {
         auto quantile_agg = static_cast<quantile_aggregation const&>(agg);
         CUDF_EXPECTS(quantile_agg._quantiles.size() == 1,
                      "Reduction quantile accepts only one quantile value");
-        auto sorted_indices = sorted_order(table_view{{col}}, {}, {null_order::AFTER}, stream);
+        auto current_mr = rmm::mr::get_current_device_resource();
+        auto sorted_indices =
+          sorted_order(table_view{{col}}, {}, {null_order::AFTER}, stream, current_mr);
```
Comment (ttnghia) [inline cpp/src/reductions/reductions.cpp:88]: Why the default stream is used to compute the temporary results? I thought that only the default resource should be used in such cases while only one (input) stream is used for computing everything, isn't it?

Comment (davidwendt) [reply cpp/src/reductions/reductions.cpp:88]: Sorry, I don't understand your question. The same `stream` is being used for all computations here.

Comment (ttnghia) [reply cpp/src/reductions/reductions.cpp:88]: Sorry, I wrongly read `current_mr` as `current_stream` (default stream). My brain stopped working today. Please just ignore it.

**Review by ttnghia — APPROVED** [review]

**Review by PointKernel — APPROVED** [review]

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**